specs/data_structures.md

```diff
@@ -496,19 +501,32 @@ One unified [Sparse Merkle Trees](#sparse-merkle-tree) is maintained for both ac
 | `balance`        | `uint64`                  | Coin balance.                                                                               |
 | `nonce`          | `uint64`                  | Account nonce. Every outgoing transaction from this account increments the nonce.           |
 | `isValidator`    | `bool`                    | Whether this account is a validator or not. Mutually exclusive with `isDelegating`.         |
-| `validatorInfo`  | [Validator](#validator)   | _Optional_, only if `isValidator` is set. Validator info.                                   |
 | `isDelegating`   | `bool`                    | Whether this account is delegating its stake or not. Mutually exclusive with `isValidator`. |
 | `delegationInfo` | [Delegation](#delegation) | _Optional_, only if `isDelegating` is set. Delegation info.                                 |
 
-In the accounts tree, accounts (i.e. leaves) are keyed by the [hash](#hashdigest) of their [address](#address).
+In the accounts subtree, accounts (i.e. leaves) are keyed by the [hash](#hashdigest) of their [address](#address). The first byte is then replaced with `ACCOUNTS_SUBTREE_ID`.
```

Comment:
Effectively, this means that the hash used in the subtree returns 31 bytes.

Reply:
Yes.

Reply:
If we say it's the hash function, this is easier to formalize. e.g. we can say the key for an account in the set `Accs` tree is computed as:
`key := subtree_id || hash_st(address), `
where `hash_st` is any cryptographic hash function `hash_st: Accs -> {0,1}^31`

Reply:
Hmm. I'm not really a fan because that would mean we need 1) a different hashing function for each subtree (not great, not terrible) and 2) to slice off the first byte of every single hashing operation. That cost will add up quickly, especially if doing proofs in smart contracts (which will need to be done even in the absence of fraud proofs---Merkle proofs need to be provided for transaction and message inclusion to Ethereum contracts that use LL for data availability.

I'd much prefer just changing how the keys are calculated, which is a one-time calculation per leaf.

Reply:
IMO It's just another way to writing down the same thing though (of course in 2) the slicing off could be done more efficiently by replacing the 1st byte instead; but this is just an implementation detail).